### PR TITLE
Clean up when a driver disconnects.

### PR DESCRIPTION
**Pull request by robertnishihara**
This PR should address #376.

- When a driver exits, this is detected by the local scheduler that the driver was connected to.
- The local scheduler publishes a message saying that the driver died.
- All local schedulers receive the message and clean up their data structures (e.g., kill any actors for that driver, kill any workers running tasks for that driver, and remove tasks for that driver from their internal queues).
- The monitor also receives the message and releases any GPU resources in Redis that are recorded as being used by actors for that driver.

This PR also expands the multi node docker tests to support tests with multiple drivers, and it establishes the convention of having a single test script that branches on a `driver_index` to run different driver scripts. It also uses Redis as a means to coordinate when the drivers (e.g., make one driver run only after the other has finished).

### CODE REVIEW

File: src/local_scheduler/local_scheduler_algorithm.cc [hunk 1]
```diff
@@ -1141,6 +1144,61 @@ void handle_object_removed(LocalSchedulerState *state,
...
+    }
+    /* If there are no more dependent tasks for this object, then remove the
+     * ObjectEntry. */
+    if (it->second.dependent_tasks.size() == 0) {
+      it = algorithm_state->remote_objects.erase(it);
+    } else {
+      it++;
+    }
+  }
+  // std::unordered_map<ObjectID, ObjectEntry, UniqueIDHasher> remote_objects;
```
Comment (pcmoritz) [inline src/local_scheduler/local_scheduler_algorithm.cc]: Remove these

File: src/common/state/redis.cc [hunk 2]
```diff
@@ -1124,15 +1209,20 @@ void redis_actor_notification_table_subscribe_callback(redisAsyncContext *c,
     redisReply *payload = reply->element[2];
     ActorNotificationTableSubscribeData *data =
         (ActorNotificationTableSubscribeData *) callback_data->data;
-    ActorInfo info;
-    /* The payload should be the concatenation of these two structs. */
-    CHECK(sizeof(info.actor_id) + sizeof(info.local_scheduler_id) ==
+    /* The payload should be the concatenation of three IDs. */
```
Comment (pcmoritz) [inline src/common/state/redis.cc]: Can we make this a flatbuffer message?

Comment (robertnishihara) [inline src/common/state/redis.cc]: I tried actually. The problem is that it's created from Python, and it doesn't seem like creating flatbuffer messages in Python works.

Comment (robertnishihara) [inline src/common/state/redis.cc]: Ok, this should be done now.

Comment (robertnishihara) [inline src/common/state/redis.cc]: Reverted, because it didn't work in Python 2.7.

**Pull request merged**